### ISSUE TITLE
ar model maxlag default setting

### ISSUE THREAD

**Issue by gardenia22** [statsmodels/statsmodels#2415]
In ar model, the default of ``maxlag`` is ``round(12*(nobs/100.)**(1/4.))``
However, when input array is less than 6, the default ``maxlag`` would be larger than ``nobs``, which causes error to fit.

```
>>> from statsmodels.api import tsa
>>> a = [1,2,3,4,5,6]
>>> ar_mod = tsa.AR(a)
>>> ar_res = ar_mod.fit()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/statsmodels-0.6.1-py2.7-macosx-10.10-intel.egg/statsmodels/tsa/ar_model.py", line 549, in fit
    X = self._stackX(k_ar, trend)  # sets self.k_trend
  File "/Library/Python/2.7/site-packages/statsmodels-0.6.1-py2.7-macosx-10.10-intel.egg/statsmodels/tsa/ar_model.py", line 391, in _stackX
    X = lagmat(endog, maxlag=k_ar, trim='both')
  File "/Library/Python/2.7/site-packages/statsmodels-0.6.1-py2.7-macosx-10.10-intel.egg/statsmodels/tsa/tsatools.py", line 328, in lagmat
    raise ValueError("maxlag should be < nobs")
ValueError: maxlag should be < nobs
```

**Comment by josef-pkt** [comment]
It looks like this was supposed to be fixed in ff49a8e020ca5179bb65742a7ef68753572924b5
closed issue for ARMA #1146

**Comment by bashtage** [comment]
@josef-pkt Looks like this one is closable

**Comment by josef-pkt** [comment]
Thanks, yes fixed in #1146 to use nobs-1 as maxlag in this case
